examples/microphone_nextjs/pages/index.tsx

```diff
@@ -24,8 +35,9 @@ export default function Main({ jwt }: MainProps) {
   const rtSessionRef = useRef(new RealtimeSession(jwt));
 
   // Get devices using our custom hook
-  const [requestDevices, setRequestDevices] = useState<boolean>(false);
-  const [devices, denied] = useAudioDevices(requestDevices);
+  const devices = useAudioDevices();
+  const denied = useAudioDenied();
+  const requestDevices = useRequestDevices();
 
   // useEffect listens for changes in devices
   // It sets a default deviceId if no valid deviceId is already set
```

Comment:
I think the useEffect below this can easily be removed too - I'll take a look shortly, can either go in as part of this PR or as a follow up

Reply:
(as it falls under "Updating state based on props or state")

https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
Yeah, I think actually that use effect is needed regardless as MediaRecorder automatically uses the default device if the deviceId value provided is falsy